cli/test-utils/src/script.rs

```diff
@@ -0,0 +1,153 @@
+use ethers::{
+    abi::Address,
+    prelude::{Http, Middleware, NameOrAddress, Provider, U256},
+    utils::hex,
+};
+
+use std::{collections::BTreeMap, path::Path, str::FromStr};
+
+use crate::TestCommand;
+
+/// A helper struct to test forge script scenarios
+pub struct ScriptTester {
+    pub accounts_pub: Vec<Address>,
+    pub accounts_priv: Vec<String>,
+    pub provider: Provider<Http>,
+    pub nonces: BTreeMap<u32, U256>,
+    pub cmd: TestCommand,
+    pub err: bool,
+}
+
+impl ScriptTester {
+    pub fn new(mut cmd: TestCommand, port: u16, current_dir: &Path) -> Self {
+        ScriptTester::link_testdata(current_dir).unwrap();
+        cmd.set_current_dir(current_dir);
+
+        let target_contract = current_dir.join("src/Broadcast.t.sol").to_string_lossy().to_string();
+        let url = format!("http://127.0.0.1:{port}");
+
+        cmd.args([
+            "script",
+            "-r",
+            "ds-test/=lib/",
+            target_contract.as_str(),
+            "--root",
+            current_dir.to_str().unwrap(),
+            "--fork-url",
+            url.as_str(),
+            "-vvvvv",
+        ]);
+
+        ScriptTester {
+            accounts_pub: vec![
+                Address::from_str("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266").unwrap(),
+                Address::from_str("0x70997970C51812dc3A010C7d01b50e0d17dc79C8").unwrap(),
+                Address::from_str("0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC").unwrap(),
+            ],
+            accounts_priv: vec![
+                "ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80".to_string(),
+                "59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d".to_string(),
+                "5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a".to_string(),
```

Comment:
should we instead store the priv keys as `Vec<u8>`?

Reply:
they're only used as strings anyway, so i feel like it should be fine?